pkg/controller/core.oam.dev/v1alpha2/applicationrollout/applicationrollout_controller.go

```diff
@@ -94,29 +94,39 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (res reconcile.Result, retErr e
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 	klog.InfoS("Start to reconcile ", "appRollout", klog.KObj(&appRollout))
-	if len(appRollout.Status.RollingState) == 0 {
-		appRollout.Status.ResetStatus()
-	}
-	targetAppRevisionName := appRollout.Spec.TargetAppRevisionName
-	sourceAppRevisionName := appRollout.Spec.SourceAppRevisionName
 
 	// handle app Finalizer
 	doneReconcile, res, retErr := r.handleFinalizer(ctx, &appRollout)
 	if doneReconcile {
 		return res, retErr
 	}
 
+	reconRes, err := r.DoReconcile(ctx, &appRollout)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	return reconRes, r.updateStatus(ctx, &appRollout)
+}
+
+// DoReconcile is real reconcile logic for appRollout
```

Comment:
```suggestion
// DoReconcile is real reconcile logic for appRollout. !!! Note the AppRollout object should not be updated in this function as it could be logically used in Application reconcile loop which does not have real AppRollout object.
```